internal/transform/transform_test.go

```diff
@@ -199,3 +199,47 @@ func TestFullTransform(t *testing.T) {
 		})
 	}
 }
+
+func TestAnnotation(t *testing.T) {
+	tests := []struct {
+		name   string
+		source string
+		want   string
+	}{
+		{
+			name:   "basic",
+			source: `<div>Hello world!</div>`,
+			want:   `<div data-astro-source-file="/test.astro">Hello world!</div>`,
+		},
+		{
+			name:   "no components",
+			source: `<Component>Hello world!</Component>`,
+			want:   `<Component>Hello world!</Component>`,
+		},
+		{
+			name:   "injects root",
+			source: `<html></html>`,
+			want:   `<html data-astro-source-root="file://"></html>`,
```

Comment:
We inject `data-astro-source-root={transformOpts.ProjectRoot}` to the `html` element